torch/lib/c10d/ProcessGroup.cpp

```diff
@@ -145,31 +146,25 @@ void ProcessGroup::Work::abort() {
 }
 
 c10::intrusive_ptr<c10::ivalue::Future> ProcessGroup::Work::getFuture() {
-  TORCH_CHECK(false, "ProcessGroup::Work::getFuture not implemented.")
+  return future_;
+}
+
+void ProcessGroup::Work::finish(c10::IValue value) {
+  future_->markCompleted(value);
 }
 
 void ProcessGroup::Work::finish(std::exception_ptr exception) {
-  std::unique_lock<std::mutex> lock(mutex_);
-  completed_ = true;
-  exception_ = exception;
-  if (recordFunctionEndCallback_) {
-    recordFunctionEndCallback_();
```

Comment:
This might break distributed collectives profiling, or is it now handled elsewhere?